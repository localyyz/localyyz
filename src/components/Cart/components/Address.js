import React from "react";
import {
  View, StyleSheet, Text, TouchableOpacity
} from "react-native";
import {
  Colours, Sizes, Styles
} from "localyyz/constants";

// custom
import CartField from "./CartField";

// third party
import * as Animatable from "react-native-animatable";
import LinearGradient from "react-native-linear-gradient";
import EntypoIcon from "react-native-vector-icons/Entypo";

export default class Address extends React.Component {
  render() {
    return (
      <CartField
        icon="home"
        onPress={this.props.onPress}
        onIconPress={this.props.onIconPress}>
        {this.props.children || (
          <View style={styles.address}>
            <Text style={[
              Styles.Text, Styles.Emphasized, Styles.SmallText,
              styles.addressLine]}>
              {this.props.address.shortAddress}
            </Text>
            <Text style={[
              Styles.Text, Styles.Terminal, Styles.Subdued,
              styles.addressDetails]}>
              {this.props.address.extendedAddress}
            </Text>
          </View>)}
        <TouchableOpacity onPress={this.props.onActionPress}>
          <LinearGradient
            colors={[Colours.Foreground, Colours.Transparent]}
            locations={[0.8, 1]}
            start={{x: 1}}
            end={{x: 0}}
            style={styles.iconContainer}>
            <EntypoIcon
              name={this.props.buttonIcon || "edit"}
              size={Sizes.IconButton / 2}
              color={this.props.buttonColor || Colours.Text}
              style={Styles.IconOffset} />
          </LinearGradient>
        </TouchableOpacity>
      </CartField>
    );
  }
}

const styles = StyleSheet.create({
  address: {
    flex: 1,
    flexWrap: "nowrap",
    flexDirection: "row",
    alignItems: "center",
    marginLeft: Sizes.InnerFrame
  },

  addressDetails: {
    textAlign: "left",
    marginLeft: Sizes.InnerFrame / 2,
    paddingRight: Sizes.InnerFrame / 2
  },

  iconContainer: {
    marginLeft: -Sizes.InnerFrame * 2.2
  }
});
