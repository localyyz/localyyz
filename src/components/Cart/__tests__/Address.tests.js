import React from "react";
import Address from "../components/Address";
import Renderer from "react-test-renderer";

const props = {
  addressLine: "Localyyz",
  address: {
    shortAddress: "180 John Street",
    extendedAddress: "Toronto, Ontario, Canada"
  },
  buttonColor: "#000000"
};
describe("Address", () => {
  it("Address: should render properly", () => {
    let rendered = Renderer.create(<Address {...props} />).getInstance();
    expect(rendered.refs.address.props.children[0].props.children).toBe(
      "180 John Street"
    );
    expect(rendered.refs.address.props.children[1].props.children).toBe(
      "Toronto, Ontario, Canada"
    );
    expect(rendered.refs.addressEditIcon.props.name).toBe("edit");
    expect(rendered.refs.addressEditIcon.props.color).toBe("#000000");

    expect(rendered.refs.addressRemoveIcon.props.name).toBe("trash");
    expect(rendered.refs.addressRemoveIcon.props.color).toBe("#000000");
  });

  it("Address: should render properly(snapshot)", () => {
    let rendered = Renderer.create(<Address {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
