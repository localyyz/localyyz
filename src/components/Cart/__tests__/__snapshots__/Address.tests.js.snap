// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address Address: should render properly(snapshot) 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginVertical": 1,
        "paddingHorizontal": 7.5,
        "paddingVertical": 15,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "marginLeft": 7.5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0)",
            "color": "rgb(30, 30, 30)",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        180 John Street
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0)",
            "color": "#999999",
            "fontSize": 16,
            "fontWeight": "300",
            "marginLeft": 3.75,
            "paddingRight": 7.5,
            "textAlign": "left",
          }
        }
      >
        Toronto, Ontario, Canada
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <BVLinearGradient
        colors={
          Array [
            4294967295,
            16777215,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        locations={
          Array [
            0.8,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 1,
            "y": 0,
          }
        }
        style={
          Object {
            "marginHorizontal": 7.5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 10,
              },
              Object {
                "marginRight": 6,
              },
              Object {
                "fontFamily": "Entypo",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </BVLinearGradient>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <BVLinearGradient
        colors={
          Array [
            4294967295,
            16777215,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        locations={
          Array [
            0.8,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 1,
            "y": 0,
          }
        }
        style={
          Object {
            "marginHorizontal": 7.5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 10,
              },
              Object {
                "marginRight": 6,
              },
              Object {
                "fontFamily": "Entypo",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </BVLinearGradient>
    </View>
  </View>
</View>
`;
